### ISSUE TITLE
Issue with gradle configurations like 'implementation', 'testImplementation'

### ISSUE THREAD

**Issue by vlipovetskii** [johndevs/gradle-vaadin-plugin#506]
I strive to leverage new gradle configurations like 'implementation', 'testImplementation', since 'compile' and 'testCompile' are deprecated. It looks that vaadin-plugin doesn't support the new gradle configurations, since gretty fails to launch war. I have discovered in the GradleVaadinPlugin.groovy fragment, that I think, confirms my suggestion:
`…
// Needed so bootRepackage can include all dependencies in Jar
                    conf.extendsFrom(
                            project.configurations['compile'],
                            project.configurations['runtime'],
                            project.configurations[CONFIGURATION_PUSH],
                            project.configurations[CONFIGURATION_CLIENT_COMPILE]
                    )
…

`
As a workaround I temporarily added 'compile' and 'testCompile'.

Possible, that issue of @mvysny (https://github.com/johndevs/gradle-vaadin-plugin/issues/501) and mine are connected ? I leverage karibu-dsl by @mvysny.

**Comment by johndevs** [comment]
Yes, right now the new implementation configurations are not supported. 

Unfortunately, it is a breaking change which means that either we have to release a non-backwards compatible plugin or continue to use compile configurations as they work in both new and old versions of Gradle.
